src/FieldWarning/Assets/Terrain/Scripts/TerrainMap.cs

```diff
@@ -43,32 +45,39 @@ public class TerrainMap
     public const float BRIDGE_WIDTH = 3f * TerrainConstants.MAP_SCALE;
     public const float BRIDGE_HEIGHT = 1.0f; // temporary
 
-    public const string HEIGHT_MAP_SUFFIX = "_sample_water.dat";
+    public const string HEIGHT_MAP_SUFFIX = "_map_terrain_cache.dat";
     private readonly string _HEIGHT_MAP_PATH;
 
     public readonly Vector3 MapMin, MapMax, MapCenter;
 
+    // internal variable used for main thread notification
+    private bool _processing = false;
+
     private byte[,] _map;
     private int _mapSize;
     private float _terrainSpacingX, _terrainSpacingZ;
 
     // 2D array of terrain pieces for quickly finding which piece is at a given location
     private Terrain[,] _terrains;
 
-    private Loading _loader;
+    //private Loading _loader;
 
     public readonly float WATER_HEIGHT;
 
     private GameObject[] _bridges;
     private ERModularRoad[] _roads;
+
+    public int TerrainId;
+
     List<Vector3> _treePositions = new List<Vector3>();
     List<Vector3> _bridgePositions = new List<Vector3>();
 
     // this is only needed for map testing
     private byte[,] originalTestMap = null;
```

Comment:
generates a warning about unused variable

Reply:
hmm this one is definitely used. I'm not saying any unused warning for that.